packages/firestore/src/core/firestore_client.ts

```diff
@@ -270,13 +270,18 @@ export class FirestoreClient {
         this.asyncQueue,
         serializer
       );
-      this.sharedClientState = new WebStorageSharedClientState(
-        this.asyncQueue,
-        this.platform,
-        storagePrefix,
-        this.clientId,
-        user
-      );
+      if (WebStorageSharedClientState.isAvailable(this.platform)) {
+        this.sharedClientState = new WebStorageSharedClientState(
+          this.asyncQueue,
+          this.platform,
+          storagePrefix,
+          this.clientId,
+          user
+        );
+      } else {
+        debug(LOG_TAG, 'Starting Persistence in non multi-tab mode');
+        this.sharedClientState = new MemorySharedClientState();
```

Comment:
I think this is something we can eventually support, but we should decide if this extra complexity is worth having when customers start asking for it. I added an assert for MOCK_PERSISTENCE since it will make our lives easier in the short run.